src/components/screens/SwapScreen/hooks/useSwapTransaction.ts

```
@@ -113,9 +115,22 @@ export const useSwapTransaction = ({
       if (!transactionHash) {
         throw new Error("Failed to submit transaction");
       }
+
+      analytics.trackSwapSuccess({
+        sourceAsset: sourceBalance?.tokenCode || NATIVE_TOKEN_CODE,
+        destAsset: destinationBalance?.tokenCode || NATIVE_TOKEN_CODE,
```

Should we fallback to something more explicit here instead of NATIVE_TOKEN_CODE? E.g. "Unknown Asset"

I'm afraid it could potentially pollute our metrics with misleading data.

Or maybe we could throw an error in case it doesn't make sense to execute a swap without the source or destination assets. I see that the buildSwapTransaction requires them.

---

you're right, we are now throwing the error before this so we won't have this misleading data